src/req.h

```diff
@@ -27,8 +27,11 @@ typedef struct {
   size_t max_size;
   bool dnssec;
 
+  // For Unbound
+  char namestr[HSK_DNS_MAX_NAME_STRING];
```

Comment:
Instead of adding this here, I believe it's more suitable to have it allocated only in `hsk_rs_worker_resolve`, where it's actually needed. There's no need to carry full string version just for unbound call.

Reply:
great call thanks, added in 1fffc2b831cdd53bc14c46d18122556f54848912